Rename 'not_differentiable' to 'non_differentiable'.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19431 Get rid of unnecessary matches_jit_signature: True specifications.
* #19430 Make one_hot non-differentiable.
* #19429 Remove 'BoolTensor', 'IndexTensor' from frontend specifications.
* #19428 Have _embedding_bag_dense_backward match JIT signature.
* #19427 Have embedding_dense_backward match JIT signature.
* #19426 Hook up non_differentiability in derivatives.yaml when no autograd function is generated.
* #19425 Move non_differentiable_arg_names from autograd functions to differentiability_info.
* #19424 Stop generating autograd functions for derivatives.yaml entries that only specify output differentiability.
* **#19272 Rename 'not_differentiable' to 'non_differentiable'.**

because it flows better.  Also rename 'not_differentiable_args_names' to 'non_differentiable_arg_names' because arguments isn't plural there.

Differential Revision: [D15003381](https://our.internmc.facebook.com/intern/diff/D15003381)

gchanan committed:
Rename 'not_differentiable' to 'non_differentiable'.

because it flows better.  Also rename 'not_differentiable_args_names' to 'non_differentiable_arg_names' because arguments isn't plural there.

gchanan committed:
Update on "Rename 'not_differentiable' to 'non_differentiable'."

Rename 'not_differentiable' to 'non_differentiable'.

because it flows better.  Also rename 'not_differentiable_args_names' to 'non_differentiable_arg_names' because arguments isn't plural there.

gh-metadata: pytorch pytorch 19272 gh/gchanan/16/head

cpuhrsch reviewed (approved):
not non-disagreeable

gchanan committed:
Update on "Rename 'not_differentiable' to 'non_differentiable'."

Rename 'not_differentiable' to 'non_differentiable'.

because it flows better.  Also rename 'not_differentiable_args_names' to 'non_differentiable_arg_names' because arguments isn't plural there.

gh-metadata: pytorch pytorch 19272 gh/gchanan/16/head

cpuhrsch commented on tools/autograd/load_derivatives.py:
```diff
@@ -27,6 +27,7 @@ def load_derivatives(path, declarations):
 
     return autograd_functions
 
+
```
Are you sure?

gchanan replied:
let's see what flake8 says.

cpuhrsch replied:
I have strong opinions on this that I'd like to talk to you in person about.

gchanan committed:
Update on "Rename 'not_differentiable' to 'non_differentiable'."

Rename 'not_differentiable' to 'non_differentiable'.

because it flows better.  Also rename 'not_differentiable_args_names' to 'non_differentiable_arg_names' because arguments isn't plural there.

No codegen difference.

gh-metadata: pytorch pytorch 19272 gh/gchanan/16/head

gchanan commented:
@pytorchbot retest this please.
